services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/functionaltests/document/APISurfaceAreaReview.java

```diff
@@ -0,0 +1,401 @@
+                                                                                    AttributeConverterProvider.defaultProvider())
+                                                       .build());
+
+
+        /**
+         * Creating SCHEMA with Primary and secondary keys
+         */
+
+        DocumentTableSchema tableSchema = TableSchema.fromDocumentSchemaBuilder()
+                                                     .primaryKey("sampleHashKey", AttributeValueType.S)
+                                                     .sortKey("sampleSortKey", AttributeValueType.S)
+                                                     .attributeConverterProviders(CustomAttributeConverterProvider.create(),
+                                                                                  AttributeConverterProvider.defaultProvider())
+                                                     .build();
+
+        assertThat(tableSchema.attributeNames()).isEqualTo(Arrays.asList("sampleHashKey", "sampleSortKey"));
+
+
+        /**
+         * Creating SCHEMA with NO Primary and secondary keys
+
+         */
+
+        DocumentTableSchema tableSchemaNoKey = TableSchema.fromDocumentSchemaBuilder()
+                                                          .attributeConverterProviders(CustomAttributeConverterProvider.create(),
+                                                                                       AttributeConverterProvider.defaultProvider())
+                                                          .build();
+        DynamoDbTable<EnhancedDocument> tableWithNoPrimaryDefined = enhancedClient.table(tableName, tableSchemaNoKey);
+        // User cannot create a table
+
+        assertThatExceptionOfType(IllegalArgumentException.class)
+            .isThrownBy(
+                () -> tableWithNoPrimaryDefined.createTable()
+            )
+            .withMessage("Attempt to execute an operation that requires a primary index without defining any primary key "
+                         + "attributes in the table metadata.");
+
+
+    }
+
+    /**
+     * Question 1 : Should we add API to input {@link TableMetadata} in DocumentTableSchema.Builder ?
+     * Note that  {@link TableMetadata} cannot be set by user on DocumentTableSchema.Builder.
+     *
+     */
+
+
+    /**
+     * Creating implementation of {@link EnhancedDocument} interface
+     */
+
+    /**
+     * Case 1: Creating simple Enhanced document using default types like String , Number , bytes , boolean.
+     */
+
+
+    @Test
+    void enhancedDocDefaultSimpleAttributes() {
+        /**
+         * No attributeConverters supplied, in this case it uses the {@link DefaultAttributeConverterProvider} and does not error
+         */
+        EnhancedDocument simpleDoc = EnhancedDocument.builder()
+                                                     .addString("HashKey", "abcdefg123")
```

Comment:
nit: s/add/put/ to match our naming conventions for map-like structures? "add" would be for lists/collections.